### DISCUSSION TITLE
v3.7.0

### DISCUSSION THREAD

**Discussion by edmondas** [poweradmin/poweradmin#541] [General]
## What's Changed
The focus of this release was on removing hooks and refactoring code that limits future enhancements. There is now initial support for DNSSEC over the PowerDNS API.

_Code Quality Improvements:_
- Fixed issues with quotes and improved translation accuracy.
- Removed unused code and split DNSSEC class for API-based implementation.
- Simplified code structure, updated class names, and removed autogenerated comments.

_Dependency and Configuration Management:_
- Updated project dependencies and development dependencies.
- Implemented initial API DNSSEC and adjusted parameter naming conventions.
- Enabled API calls in legacy code using fake configuration.
- Updated Docker base image, Symfony dependencies, and application version.
- Improved error handling, response management, and configuration management.

_Refactoring and Optimization:_
- Updated LDAP user management and introduced return types.
- Removed plugin dependencies and streamlined OOP implementation.
- Reduced directory nesting and standardized include syntax.
- Simplified locale setting and internationalization implementation.
- Transformed installer helper utils into a class and improved record name sanitization.

_Security and Testing:_
- Enhanced PHPUnit configuration for better error and warning verbosity.

_Miscellaneous Updates:_
- Passed database instance to legacy functions and removed global variables.
- Fixed SQLite backend issues preventing zone updates.
- Supported unquoted TXT records with improved validation checks.
- Introduced new locales and updated translations.
- Added quotes to reverse_search_string for improved search functionality.

**Full Changelog**: https://github.com/poweradmin/poweradmin/compare/v3.6.1...v3.7.0

<hr /><em>This discussion was created from the release <a href='https://github.com/poweradmin/poweradmin/releases/tag/v3.7.0'>v3.7.0</a>.</em>

**Comment by edmondas** [comment]
**How to configure DNSSEC over API**

Configuration like this should be added:
```
$pdnssec_use = true;
$pdns_api_url = 'http://127.0.0.1:8081'; // Replace the PowerDNS API URL with your own
$pdns_api_key = '65b5dea9-dcfd-4a81-89dd-9f7a1afb0f10'; // Enter your API key here.
```

**Comment by danfratamico** [comment]
There are many missing files in the "inc" folder. Was this intentional? Are they not being used anymore?

**Reply by edmondas** [reply]
This was intentional, all functionality has been relocated to related classes, and the 'inc' folder is only for local and default configuration settings.

**Reply by edmondas** [reply]
Now the inc folder will look pretty simple:
<img width="226" alt="image" src="https://github.com/poweradmin/poweradmin/assets/9033112/3e10244d-7583-4ef0-9e15-5af8301293cf">
